pkg/cloud/services/iamauth/reconcile.go

```diff
@@ -49,7 +50,8 @@ func (s *Service) ReconcileIAMAuthenticator(ctx context.Context) error {
 		return fmt.Errorf("getting aws-iam-authenticator backend: %w", err)
 	}
 
-	roleARN := fmt.Sprintf("arn:aws:iam::%s:role/nodes%s", accountID, iamv1.DefaultNameSuffix)
+	partition := system.GetPartitionFromRegion(s.scope.Region())
+	roleARN := fmt.Sprintf("arn:%s:iam::%s:role/nodes%s", partition, accountID, iamv1.DefaultNameSuffix)
```

Comment:
A related issue here is that we use default name suffix, instead of respecting the user-defined role. That is fixed in #4011, which is being reviewed as I write this.